### DISCUSSION TITLE
How can I get this to work with Unity WebGL?

### DISCUSSION THREAD

**Discussion by ckline5** [melanchall/drywetmidi#269] [Q&A] [answered]
Hello,

I am building a Unity WebGL project which includes some MIDI stuff using DryWetMidi. Everything works great while running in the Unity editor but when I build it and run in a web browser, I get the following error:

"
**Release.framework.js.gz:10 DllNotFoundException: Unable to load DLL 'Melanchall_DryWetMidi_Native32'. Tried the load the following dynamic libraries:** 
"

It doesn't tell me any dynamic libraries it tried to load; it just errors with that message. This error occurs immediately upon calling InputDevice.GetByName().

I tried creating a brand-new Unity project with DryWetMidi to replicate the issue, and it still persists even in a fresh project.

I am using the latest version of DryWetMidi (v7.0.0) from the Unity asset store.

Any help on this is greatly appreciated.

**Comment by melanchall** [comment, marked as answer]
Hi,

I'm afraid it's not possible to use the Devices API of the library within WebGL. In fact, I didn't even test such an integration.

**Reply by ckline5** [reply]
That is disappointing to hear. Are there any plans to add WebGL support?

**Reply by melanchall** [reply]
I have such plans, but unfortunately can't provide you any dates :-(
